scripts/gitlab/test-linux.sh

```diff
@@ -14,4 +14,4 @@ rustup default $1
 rustup show
 
 echo "________Running Parity Full Test Suite________"
-time cargo test $OPTIONS --features "$FEATURES" --locked --all --target $CARGO_TARGET --verbose --color=always -- --test-threads $THREADS
+CARGO_INCREMENTAL=0 RUSTFLAGS="-C opt-level=3 -C overflow-checks=off -C debuginfo=2" time cargo test $OPTIONS --features "$FEATURES" --locked --all --target $CARGO_TARGET --verbose --color=always -- --test-threads $THREADS
```

Comment:
why disable overflow-checks for tests? they can point to a bug more precisely
also, not introduced by this PR, but shall we remove `--color=always`?

Reply:
It's the fruit of some sleuthing as to what `--release` actually means and `overflow-checks=off` is one of the implied settings.

Reply:
you mean more compiled artifacts can be reused if this option is set?

Reply:
But `overflow-checks` didn't seem to work as a flag, did  it?
What's wrong with `--color=always`? It was set explicitly for readability of logs in CI. Because somewhy if you set `--verbose`, the `--color` defaults to `never`.

Reply:
> But `overflow-checks` didn't seem to work as a flag, did it?

It **does** work as a flag. What is odd about it is that `overflow-checks = false` is valid in `Cargo.toml` but **not** as a commandline opt (i.e. `-C overflow-checks=false` is an error).

> What's wrong with --color=always

I think @ordian wants the escape codes out of the raw logs. I'm indifferent.

Reply:
So it should be `-C overflow-checks=off` to work as a flag?

Yes, `--color` shits codes in raw logs. Btw, we could get rid of `--verbose` if it's not needed any more.

Reply:
> So it should be `-C overflow-checks=off` to work as a flag?

Yes.

> Btw, we could get rid of --verbose if it's not needed any more.

One thing at a time. :)